src/main/java/gov/usgs/earthquake/indexer/ExtentIndex.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Extent Index
+ */
+package gov.usgs.earthquake.indexer;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Types;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.sqlite.SQLiteException;
+
+/**
+ * ExtentIndex is a type of JDBCProductIndex that can also send updates to the extentSummary table.
+ */
+public class ExtentIndex extends JDBCProductIndex {
+
+  private static final Logger LOGGER = Logger.getLogger(ExtentIndex.class.getName());
+  public static final String EXTENT_TABLE = "extentSummary";
+  public static final String EXTENT_INDEX_ID = "productid";
+  public static final String EXTENT_START_TIME = "starttime";
+  public static final String EXTENT_END_TIME = "endtime";
+  public static final String EXTENT_MAX_LAT = "maxlatitude";
+  public static final String EXTENT_MIN_LAT = "minlatitude";
+  public static final String EXTENT_MAX_LONG = "maxlongitude";
+  public static final String EXTENT_MIN_LONG = "minlongitude";
+
+  public ExtentIndex() throws Exception {
+    super();
+  }
+
+  /**
+   * Queries extentSummary table for the largest index id.
+   * 
+   * @throws Exception if something goes wrong with database transaction
+   */
+  public long getLastExtentIndexId() throws Exception{
+    long lastIndex;
+
+    //Prepare statement
+    Connection connection = connect();
+    PreparedStatement getLastIndex;
+    try {
+      getLastIndex = connection.prepareStatement(
+              "SELECT MAX(" + EXTENT_INDEX_ID +
+                      ") AS " + EXTENT_INDEX_ID +
+                      " FROM " + EXTENT_TABLE);
+    } catch (SQLiteException e) {
+      throw new SQLiteException(e.getMessage() + ". SQL query was: SELECT MAX(" + EXTENT_INDEX_ID + ") AS " + EXTENT_INDEX_ID + " FROM " + EXTENT_TABLE, e.getResultCode());
+    }
+
+    //Parse Results
+    ResultSet results = getLastIndex.executeQuery();
+    if (results.next()) {
+      lastIndex = results.getLong(EXTENT_INDEX_ID);
+    } else {
+      //No index in extentSummary table
+      lastIndex = 0;
+      LOGGER.log(Level.FINEST,"[" + getName() + "] no products in extentSummary table; using index 0");
+    }
+
+    //Cleanup
+    getLastIndex.close(); //Never needed again (if callers are doing their job)
```

Comment:
Use try/finally, or try with resources to make sure prepared statement is closed

Reply:
done with try-with-resources